### PR TITLE
Fix Maven rules to support jarjar_library targets

### PR DESCRIPTION
**Pull request by vmax**
Adapt to using `jarjar_library`

### CODE REVIEW

File: maven/templates/rules.bzl [hunk 1]
```diff
@@ -171,7 +174,12 @@ def _deploy_maven_jar_impl(ctx):
     _generate_deployment_script(ctx, maven_coordinates)
 
     # there is also .source_jar which produces '.srcjar'
```
Comment (haikalpribadi) [inline maven/templates/rules.bzl]: what's the significance of this comment?

File: maven/templates/rules.bzl [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# Do not edit. deployment_rules_builder.py autogenerates this file from deployment/maven/templates/rules.bzl
+
 #
 # GRAKN.AI - THE KNOWLEDGE GRAPH
 # Copyright (C) 2018 Grakn Labs Ltd
```
Comment (haikalpribadi) [inline maven/templates/rules.bzl]: Linking the comment made in the other PR https://github.com/graknlabs/grakn/pull/4703#discussion_r241356298.

File: maven/templates/rules.bzl (hunk 1 above)
Comment (vmax) [reply maven/templates/rules.bzl]: @haikalpribadi this should give a hint why we use `.class_jar` in https://github.com/vmax/graknlabs-deployment/blob/master/maven/templates/rules.bzl#L174 as opposed to `.source_jar`
Any better way to phrase it?

File: maven/templates/rules.bzl (hunk 2 above)
Comment (vmax) [reply maven/templates/rules.bzl]: @haikalpribadi I'll remove license header for now

**Pull request merged**